plugins/modules/database/mysql/mysql_user.py

```diff
@@ -561,14 +561,14 @@ def pick(x):
         res = re.match("""GRANT (.+) ON (.+) TO (['`"]).*\\3@(['`"]).*\\4( IDENTIFIED BY PASSWORD (['`"]).+\\6)? ?(.*)""", grant[0])
         if res is None:
             raise InvalidPrivsError('unable to parse the MySQL grant string: %s' % grant[0])
-        privileges = res.group(1).split(", ")
-        privileges = [pick(x) for x in privileges]
+        privileges = res.group(1).split(",")
```

Comment:
By default, this regex won't match multiline.

Extract from https://docs.python.org/3/library/re.html:
```
[...]
The special characters are:

'.'
(Dot.) In the default mode, this matches any character except a newline. If the DOTALL flag has been specified, this matches any character including a newline.
[...]
```

I therefore suppose that the `split` + `strip` + `join` may break if a double-keyword is split in two lines, as the priv would be re-aggregated as two separate keywords.

Reply:
I don't think it should match multiple lines, mysql actually produces __multiple__ `GRANT` statements:

```
+-----------------------------------------------------------------------------------------------------------------------------------------------------+
| Grants for replication_admin@%                                                                                                                      |
+-----------------------------------------------------------------------------------------------------------------------------------------------------+
| GRANT SELECT, RELOAD, SHUTDOWN, PROCESS, FILE, SUPER, EXECUTE, REPLICATION SLAVE, REPLICATION CLIENT, CREATE USER ON *.* TO `replication_admin`@`%` |
| GRANT BACKUP_ADMIN,CLONE_ADMIN,PERSIST_RO_VARIABLES_ADMIN,SYSTEM_VARIABLES_ADMIN ON *.* TO `replication_admin`@`%` WITH GRANT OPTION                |
| GRANT INSERT, UPDATE, DELETE ON `mysql`.* TO `replication_admin`@`%` WITH GRANT OPTION                                                              |
| GRANT ALL PRIVILEGES ON `mysql_innodb_cluster_metadata`.* TO `replication_admin`@`%` WITH GRANT OPTION                                              |
| GRANT ALL PRIVILEGES ON `mysql_innodb_cluster_metadata_bkp`.* TO `replication_admin`@`%` WITH GRANT OPTION                                          |
| GRANT ALL PRIVILEGES ON `mysql_innodb_cluster_metadata_previous`.* TO `replication_admin`@`%` WITH GRANT OPTION                                     |
+-----------------------------------------------------------------------------------------------------------------------------------------------------+
```
As you can see there, the first two lines are for replication_admin

Reply:
My bad (covid got me 🙄🤒 ), when I read your PR, your long grant was on multiple lines but I didn't notice that there was no line numbers, meaning it was on a single line.
I was wondering why this old regex never crashed... I now know why 😄

Reply:
@bmalynovytch get well soon!

Reply:
Thanks ! Still 4/5 days before doing well again !

Reply:
ohh..

Reply:
Ouch, all the best and sorry for pinging you.